File: Assets/Scripts/EMPpulse.js
#pragma strict

var explodeTime : float = 3;
var startTime : float = 0;
var exploded : boolean = false;

function Start () {
	startTime = Time.time;
	exploded = false;
}

function Update () {
	if (Time.time >= startTime + explodeTime)
	{
		if (!exploded)
		{
			Pulse();
			exploded = true;
		}
	}
}

function Pulse() {
	var hits : RaycastHit[];
	
	hits = Physics.SphereCastAll(transform.position,3,transform.forward);
	for (var hit : RaycastHit in hits)
	{
		//Debug.Log(hit.collider.gameObject.name);
		if (hit.collider.transform.parent != null)
			hit.collider.transform.parent.BroadcastMessage("Scramble",SendMessageOptions.DontRequireReceiver);
		else
			hit.collider.BroadcastMessage("Scramble",SendMessageOptions.DontRequireReceiver);
	}
}

File: Assets/Scripts/JPeezy/EMPPulse.cs
using UnityEngine;
using System.Collections;

public class EMPPulse : MonoBehaviour {
	float startTime = 0;
	public float pulseTime = 2f;
	public bool pulsed = false;
	public float pulseRadius = 5f;
	// Use this for initialization
	void Start () {
		startTime = Time.time;
	}
	
	// Update is called once per frame
	void Update () {
		if (!pulsed && Time.time >= startTime + pulseTime)
		{
			Pulse();
		}
	}

	void Pulse()
	{
		Collider[] hitColliders = Physics.OverlapSphere(transform.position, pulseRadius);
		
		int i = 0;
		while (i < hitColliders.Length)
        {
        	print(hitColliders[i].name);
        	if (hitColliders[i].transform.parent != null)
            	hitColliders[i].transform.parent.gameObject.SendMessage("OnEMP",SendMessageOptions.DontRequireReceiver);
            	hitColliders[i].gameObject.SendMessage("OnEMP",SendMessageOptions.DontRequireReceiver);
           	i++;
        }
        pulsed = true;
	}
}
